tests/test_sources.py

```diff
@@ -179,3 +165,139 @@ def test_patch_sources_apply_only_unversioned(
+
+    sources._warn_for_old_patch(source_root_dir, patches_dir)
+    mock.assert_not_called()
+
+
+@patch("logging.Logger.warning")
+def test_warning_for_older_patch_same_version(mock, tmp_path: pathlib.Path):
+    # create patches dir
+    patches_dir = tmp_path / "patches_dir"
+    patches_dir.mkdir()
+
+    # create patches dir for old version of deepspeed
+    deepspeed_old_patch = patches_dir / "deepspeed-0.5.0"
+    deepspeed_old_patch.mkdir()
+    patch_file = deepspeed_old_patch / "deepspeed-0.5.0.patch"
+    patch_file.write_text("This is a test patch")
+
+    # set current source to be a new version of deepspeed
+    source_root_dir = tmp_path / "deepspeed-0.5.0"
+    source_root_dir.mkdir()
+
+    sources._warn_for_old_patch(source_root_dir, patches_dir)
+    mock.assert_not_called()
+
+
+def test_regex_dummy_package(tmp_path: pathlib.Path):
+    req_name = "foo"
+    patches_dir = tmp_path / "patches_dir"
+    patches_dir.mkdir()
+
+    lst = [
+        patches_dir / "foo-1.1.0",
+        patches_dir / "foo-bar-2.0.0",
+        patches_dir / "foo-v2.3.0",
+        patches_dir / "foo-bar-bar-v2.3.1",
+        patches_dir / "foo-bar-v5.5.5",
+        patches_dir / "foo-3.4.4",
+        patches_dir / "foo-v2.3.0.1",
+    ]
+
+    expected = [
+        patches_dir / "foo-1.1.0",
+        patches_dir / "foo-v2.3.0",
+        patches_dir / "foo-3.4.4",
+        patches_dir / "foo-v2.3.0.1",
+    ]
+
+    actual = overrides._filter_patches_based_on_req(lst, req_name)
+    assert len(expected) == len(actual)
+    assert expected == actual
+
+
+def test_regex_for_deepspeed(tmp_path: pathlib.Path):
+    req_name = "deepspeed"
+    patches_dir = tmp_path / "patches_dir"
+    patches_dir.mkdir()
+
+    lst = [
+        patches_dir / "deepspeed-1.1.0",
+        patches_dir / "deepspeed-deep-2.0.0",
+        patches_dir / "deepspeed-v2.3.0.post1",
+        patches_dir / "deepspeed-v5.5.5",
+        patches_dir / "deepspeed-3.4.4",
+        patches_dir / "deepspeed-sdg-3.4.4",
+    ]
+
+    expected = [
+        patches_dir / "deepspeed-1.1.0",
+        patches_dir / "deepspeed-v2.3.0.post1",
+        patches_dir / "deepspeed-v5.5.5",
+        patches_dir / "deepspeed-3.4.4",
+    ]
+
+    actual = overrides._filter_patches_based_on_req(lst, req_name)
+    assert len(expected) == len(actual)
+    assert expected == actual
+
+
+def test_regex_for_vllm(tmp_path: pathlib.Path):
+    req_name = "vllm"
+    patches_dir = tmp_path / "patches_dir"
+    patches_dir.mkdir()
+
+    lst = [
+        patches_dir / "vllm-1.1.0.9",
+        patches_dir / "vllm-llm-2.1.0.0",
+        patches_dir / "vllm-v2.3.5.0.post1",
+        patches_dir / "vllm-v5.5.5.1",
+    ]
+
+    expected = [
+        patches_dir / "vllm-1.1.0.9",
+        patches_dir / "vllm-v2.3.5.0.post1",
+        patches_dir / "vllm-v5.5.5.1",
+    ]
+
+    actual = overrides._filter_patches_based_on_req(lst, req_name)
+    assert len(expected) == len(actual)
+    assert expected == actual
```

Comment:
These tests should be in `test_overrides.py`

Reply:
okay, done!